DataStructures/LinkedList.java

```diff
@@ -0,0 +1,195 @@
+			// continue to next node
+			prev = currNode;
+			currNode = currNode.next;
+		}
+
+		// if we found the data in the list then delete it
+		if (currNode != null) {
+			prev.next = currNode.next;
+			System.out.println(data + " found and deleted");
+		}
+
+		// if the data is not present in the list
+		if (currNode == null) {
+			System.out.println(data + " not found");
+		}
+
+		return list;
+	}
+
+	// update a value in the linkedlist.
+	public static LinkedList update(LinkedList list, int data, int newData) {
+
+		Node currNode = list.head, prev = null;
+
+		// If head node itself holds the value to be updated
+		if (currNode != null && currNode.data == data) {
+			currNode.data = newData;
+			System.out.println(data + " found and updated");
+			return list;
+		}
+
+		// search for the data to be updated by traversing the list
+		while (currNode != null && currNode.data != data) {
+			prev = currNode;
+			currNode = currNode.next;
+		}
+
+		// if we found the data in the list then update it
+		if (currNode != null) {
+			currNode.data = newData;
+			System.out.println(data + " found and updated");
+		}
+
+		// if the data is not present in the list
+		if (currNode == null) {
+			System.out.println(data + " not found");
+		}
+
+		return list;
+	}
+
+	public static LinkedList sort(LinkedList list) {
+		Node currNode = list.head;
+		Node nextNode = list.head.next;
+		int temp;
+		while (currNode != null) {
+			while (nextNode != null) {
+				if (currNode.data > nextNode.data) {
+					temp = currNode.data;
+					currNode.data = nextNode.data;
+					nextNode.data = temp;
+				}
+				nextNode = nextNode.next;
+			}
+			currNode = currNode.next;
+			nextNode = currNode.next;
+		}
+		return list;
+	}
+
+	// Method to print the LinkedList.
+	public static void printList(LinkedList list) {
+		Node currNode = list.head;
+		System.out.print("LinkedList: ");
+
+		// Traverse through the LinkedList
+		while (currNode != null) {
+			// Print the data at current node
+			System.out.print(currNode.data + "->");
+			// Go to next node
+			currNode = currNode.next;
+		}
+		System.out.println("null");
+	}
+
+	// **************MAIN METHOD**************
+
+	public static void main(String[] args) {
+		// Create a LinkedList
+		LinkedList list = new LinkedList();
+		Boolean flag = true;
+		while (flag) {
+			System.out.println("Select an option from the menu below");
+			System.out.println("1. Insert a new node in the list");
+			System.out.println("2. Delete a node from the list");
+			System.out.println("3. Update a node in the list");
+			System.out.println("4. Sort the list");
+			System.out.println("5. Exit");
+			Scanner input = new Scanner(System.in);
+			int choice = input.nextInt();
+			if (choice < 1 || choice > 5) {
+				System.out.println("Invalid choice");
+			} else {
+				switch (choice) {
+					case 1:
+						System.out.println("Enter the value to be inserted");
+						int i_value = input.nextInt();
+						list = insert(list, i_value);
```

Comment:
Can I insert a node at an index of my choice using this code?

Reply:
No sir, we can't add a new node with this code. For that, we can create a new function that takes an index and the value of the node as input and traverse through the linked list from start and count the index with a pointer. After traversing to the (index-1) node it will insert the new node in the next of that node.

I will add that function in my code as well.